fdmi/fol_fdmi_src.c

```diff
@@ -594,6 +598,65 @@ M0_INTERNAL int m0_fol_fdmi_post_record(struct m0_fom *fom)
 	return M0_RC(rc);
 }
 
+static bool fol_fdmi_filter_match(struct m0_buf  *value,
+                                  const char    **substrings)
+{
+	struct m0_buf s;
+	m0_bcount_t   j;
+	bool          match;
+	int           i;
+
+	for (i = 0; substrings[i] != NULL; ++i) {
+		s = M0_BUF_INIT_CONST(strlen(substrings[i]), substrings[i]);
+		if (value->b_nob < s.b_nob)
+			return false;
+		match = false;
+		/* brute-force */
+		for (j = 0; j <= value->b_nob - s.b_nob; ++j) {
+			if (m0_buf_eq(&s, &M0_BUF_INIT(s.b_nob,
+			                               value->b_addr + j))) {
+				match = true;
+				break;
+			}
+		}
+		if (!match)
+			return false;
```

Comment:
if not matched, this loop only has a single iteration?
when we return, only the first element of substrings[] is compared.

Reply:
This filter requires all substrings are matched with the value, right?
if one of them is not matched, we will return false.

Reply:
Added detailed explanation to `m0_conf_fdmi_filter::ff_substrings` in https://github.com/Seagate/cortx-motr/pull/736. Please let me know if it's still not clear.